pkg/controller/atlasproject/cloud_provider_access_test.go

```diff
@@ -0,0 +1,300 @@
+package atlasproject
+
+import (
+	"context"
+	"errors"
+	"testing"
+
+	mdbv1 "github.com/mongodb/mongodb-atlas-kubernetes/pkg/api/v1"
+
+	"github.com/stretchr/testify/require"
+	"go.mongodb.org/atlas/mongodbatlas"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/customresource"
+	"github.com/mongodb/mongodb-atlas-kubernetes/pkg/controller/workflow"
+)
+
+type cloudProviderAccessClient struct {
```

Comment:
again, would move the base mock code to its own file, even in same package.

Reply:
As mentioned, will do it as a separated step.